cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1279,8 +1279,10 @@ private List<ContainerPort> getContainerPortList(KafkaPool pool) {
         }
 
         // Metrics port is enabled on all node types regardless their role
-        if (metrics.isEnabled()) {
+        if (jmxExporterMetrics.isEnabled()) {
             ports.add(ContainerUtils.createContainerPort(MetricsModel.METRICS_PORT_NAME, MetricsModel.METRICS_PORT));
+        } else if (strimziReporterMetrics.isEnabled()) {
+            ports.add(ContainerUtils.createContainerPort(StrimziMetricsReporterModel.METRICS_PORT_NAME, StrimziMetricsReporterModel.METRICS_PORT));
         }
```

Comment:
Both `jmxExporterMetrics` and `strimziReporterMetrics` can be null.

Reply:
@fvaleri does this cover it?
 ```
ports.add(jmxExporterMetrics.isEnabled()
                ? ContainerUtils.createContainerPort(MetricsModel.METRICS_PORT_NAME, MetricsModel.METRICS_PORT)
                : strimziReporterMetrics.isEnabled()
                ? ContainerUtils.createContainerPort(StrimziReporterMetricsModel.METRICS_PORT_NAME, StrimziReporterMetricsModel.METRICS_PORT)
                : null
        );
````

Reply:
You just need to add null check, there is no need to rewrite in this case.